### PR TITLE
Validate Install Plan action

### PR DESCRIPTION
**Pull request by caylahamann**
https://github.com/newrelic/newrelic-quickstarts/issues/635

we will need to add a few environment variables to the settings of the main repo
* API key 
* API url

### CODE REVIEW

File: utils/validate-install-plans.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+  ) {
+    installPlanStep {
+      id
+    }
+  }
+}
+`;
+
+const transformInstallPlanTarget = (target) => {
```
Comment (tyreer) [inline utils/validate-install-plans.js]: Could we add the comment blocks for documentation on the functions here?

File: .github/workflows/validate_install_plans.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Validate quickstarts
```
Comment (tyreer) [inline .github/workflows/validate_install_plans.yml]:
```suggestion
name: Validate install plans
```

File: utils/validate-install-plans.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  fetchPaginatedGHResults,
+  filterInstallPlans,
+} = require('./github-api-helpers');
+
+const {
+  fetchNRGraphqlResults,
+  translateMutationErrors,
+} = require('./nr-graphql-helpers');
+
+const url = pathsToSanitize[0];
```
Comment (tyreer) [inline utils/validate-install-plans.js]: We ended up naming this `GITHUB_API_URL` to distinguish it from some of the other `url` variables. Might be nice to make explicit here as well

File: utils/github-api-helpers.js [hunk 4]
```diff
@@ -62,9 +65,19 @@ const filterOutTestFiles = (files) => {
   return files.filter(({ filename }) => !MOCK_QUICKSTART_REGEXP.test(filename));
 };
 
+/**
+ * Filters out results from the Github API for changes to test files
```
Comment (tyreer) [inline utils/github-api-helpers.js]: Think this description needs updating. Should this be something about filtering out non-install plan config files?

File: utils/validate-install-plans.js (hunk 1 above)
Comment (caylahamann) [reply utils/validate-install-plans.js]: yes of course!

File: utils/validate-install-plans.js (hunk 3 above)
Comment (tyreer) [reply utils/validate-install-plans.js]:
```suggestion
const GITHUB_API_URL = passedProcessArguments[0];
```

Also renamed this array in the helpers file with 8fbf7f92

Think the sanitize naming was specific to that one file's context over here: https://github.com/tyreer/newrelic-quickstarts/blob/main/utils/sanitize_dashboards.js#L4

File: utils/__tests__/validate_install_plans.test.js [hunk 5]
```diff
@@ -30,44 +40,159 @@ const mockGithubAPIFiles = (filenames) =>
...
-    githubHelpers.filterQuickstartConfigFiles.mockReturnValueOnce(files);
+  test('succeeds with valid install plan', async () => {
+    const files = mockGithubAPIFiles([validInstallFilename]);
+    const requestBody = mockGraphqlRequestBody();
+    const response = mockGraphqlResponse();
+
+    githubHelpers.filterInstallPlans.mockReturnValueOnce(files);
+    nrGraphqlHelpers.fetchNRGraphqlResults.mockReturnValue(response);
+
+    await validateInstallPlan(files);
```
Comment (tyreer) [inline utils/__tests__/validate_install_plans.test.js]: Would it be worth asserting on the returned boolean here?
```suggestion
   const hasFailed = await validateInstallPlan(files);
```

Suppose the value would be: can we throw / log errors and still have the `process.exit` run with the default `0` argument—giving a false positive. If so, might as well assert on the return here to make sure that final block is being  addressed as we would hope. 

```
  if (hasFailed) {
    process.exit(1);
  }
```

Suppose you could could also `spyOn` the `process.exit` function if that feels more of a sure thing

File: utils/__tests__/validate_install_plans.test.js [hunk 6]
```diff
@@ -30,44 +40,159 @@ const mockGithubAPIFiles = (filenames) =>
...
+
+    githubHelpers.filterInstallPlans.mockReturnValueOnce(files);
+    nrGraphqlHelpers.fetchNRGraphqlResults.mockReturnValue(response);
 
-    validateInstallPlanIds(files);
-    expect(global.console.error).toHaveBeenCalledTimes(4);
+    await validateInstallPlan(files);
+
+    // the first error message should be the ERROR: with the filepath
+    expect(global.console.error).toHaveBeenNthCalledWith(2, error.message2);
```
Comment (tyreer) [inline utils/__tests__/validate_install_plans.test.js]: Great to see this coverage with the mock responses!

**Review by tyreer — CHANGES_REQUESTED** [review]
Looks great! A couple minor comments and questions in there. Stoked to see this coming together!

File: .github/workflows/validate_install_plans.yml (hunk 2 above)
Comment (caylahamann) [reply .github/workflows/validate_install_plans.yml]: Good catch 🤦‍♀️

File: utils/__tests__/validate_install_plans.test.js (hunk 5 above)
Comment (caylahamann) [reply utils/__tests__/validate_install_plans.test.js]: very true!!! good catch!

**Review by tyreer — APPROVED** [review]

**Comment by tyreer** [comment]
Great work! Thanks for taking this one on :D

**Pull request merged**